packages/context/src/binding-filter.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Binding, BindingTag} from './binding';
+
+/**
+ * A function that filters bindings. It returns `true` to select a given
+ * binding.
+ */
+export type BindingFilter<ValueType = unknown> = (
+  binding: Readonly<Binding<ValueType>>,
+) => boolean;
+
+/**
+ * Create a binding filter for the tag pattern
+ * @param tagPattern
+ */
+export function bindingTagFilter(tagPattern: BindingTag | RegExp) {
+  let bindingFilter: BindingFilter;
+  if (typeof tagPattern === 'string' || tagPattern instanceof RegExp) {
+    const regexp =
+      typeof tagPattern === 'string'
+        ? wildcardToRegExp(tagPattern)
+        : tagPattern;
+    bindingFilter = b => Array.from(b.tagNames).some(t => regexp!.test(t));
```

Comment:
Let's avoid using a mutable property `bindingFilter` and return the final value immediately.

```ts
return b => Array.from(b.tagNames).some(t => regexp!.test(t));
```

Also, can we use a named function please, to make error stack traces easier to read?

```ts
return function matchTagPattern(b) {
  return Array.from(b.tagNames).some(t => regexp!.test(t));
};
```

Reply:
I don't see a chance that errors will be thrown from this arrow function.

Reply:
I fixed the `return` part.